tests/Models/BeatmapsetTest.php

```diff
@@ -86,6 +88,61 @@ public function testQualify()
         $this->assertTrue($beatmapset->fresh()->isQualified());
     }
 
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testLimitedBNGNominatingBNGNominated()
+    {
+        $beatmapset = $this->createBeatmapset();
+        $beatmapset->beatmaps()->save(factory(Beatmap::class)->make());
+        factory(BeatmapMirror::class)->states('default')->create();
+
+        $user = factory(User::class)->create();
+        $user->userGroups()->create(['group_id' => UserGroup::GROUPS['bng']]);
+        $beatmapset->nominate($user);
+
+        $nominator = factory(User::class)->create();
+        $nominator->userGroups()->create(['group_id' => UserGroup::GROUPS['bng_limited']]);
+
+        priv_check_user($nominator, 'BeatmapsetNominate', $beatmapset)->ensureCan();
+    }
+
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testLimitedBNGNominatingNATNominated()
+    {
+        $beatmapset = $this->createBeatmapset();
+        $beatmapset->beatmaps()->save(factory(Beatmap::class)->make());
+        factory(BeatmapMirror::class)->states('default')->create();
+
+        $user = factory(User::class)->create();
+        $user->userGroups()->create(['group_id' => UserGroup::GROUPS['nat']]);
+        $beatmapset->nominate($user);
+
+        $nominator = factory(User::class)->create();
+        $nominator->userGroups()->create(['group_id' => UserGroup::GROUPS['bng_limited']]);
+
+        priv_check_user($nominator, 'BeatmapsetNominate', $beatmapset)->ensureCan();
+    }
+
+    public function testLimitedBNGNominatingLimitedBNGNominated()
```

Comment:
this test would fail if `Beatmapset`'s `requiredNominationCount()` were greater than 2, since the check for fullBN/NAT nominations is only relevant if the next nomination is the final one. likewise the other tests would pass but not with a useful result

not sure if this is actually an issue but the solution would be to generate `$beatmapset->requiredNominationCount() - 1` nominations first before doing that final priv_check